frontend/src/v5/ui/controls/inputs/controlledInput.component.tsx

```diff
@@ -16,32 +16,40 @@
  */
 import { Controller, ControllerRenderProps } from 'react-hook-form';
 
-type ForwardableProps = Partial<Omit<ControllerRenderProps, 'ref'> & {
+export type FormInputProps = Partial<Omit<ControllerRenderProps, 'ref'> & {
 	name: string,
 	required: boolean,
 	label: string | JSX.Element,
 	disabled: boolean,
 	className: string,
-}>;
-
-export type FormInputProps = ForwardableProps & Partial<{
 	error: boolean,
 	helperText: string,
 	inputRef: any,
 }>;
 
-type ControlledInputProps = ForwardableProps & {
-	Input: (props: FormInputProps) => JSX.Element,
+export type ControlledInputProps<T extends FormInputProps> = T & {
+	Input: (props: T) => JSX.Element,
 	name: string,
 	control?: any,
 	formError?: any,
 	defaultValue?: any,
 };
-export const ControlledInput = ({ Input, name, control, formError, ...props }: ControlledInputProps) => (
+
+// eslint-disable-next-line
+export const ControlledInput = <T,>({
```

Comment:
This is interesting. Not saying it's wrong. I'm just curious what the trailing comma is about?

Reply:
The trailing comma used this way is a special syntax that allows to use generics, but does not force them to be passed. In this case, `Input` takes some props which are dynamic, so defined by a generic. The generic type is then inferred automatically by typescript